### DISCUSSION TITLE
User settings: Apply and add symbols

### DISCUSSION THREAD

**Discussion by janedenone** [rspamd/rspamd#4083] [Q&A]
Hi,

I defined a new symbol (OWN_MAIL) with a score of -5.0 to lower the spam score of messages being sent by authenticated users in `local.d/groups.conf`:

```
symbols = { 
  "OWN_MAIL" {
    weight = -5.0;
    description = "Mail sent by authenticated user";
  }
}
```
The following syntax also found [in the documentation](https://rspamd.com/doc/configuration/metrics.html) seems to be equivalent (or at least has the same effect, I tried both):

```
symbol "OWN_MAIL" {
    weight = -5.0;
    description = "Mail sent by authenticated user";
}
```

Now in `local.d/settings.conf`, I added:

```
authenticated {
   priority = high;
   authenticated = yes;
   apply {
     OWN_MAIL = -5.0;
   }
   symbols [ "OWN_MAIL" ]
}
```

This works well, but I wonder why I have to use both the apply clause (which does not change the symbol's score as defined in `local.d/groups.conf`) *and* the symbols array. If I remove the apply clause, the symbol will not be added.

Thanks,
Jan

**Comment by gav-** [comment]
I know this is an old question, but I hit this from a google search, and found the same issue. It would appear an "apply" section is needed, even if empty for the listed symbols to work.

Here is my `local.d/settings.conf`:
```
authenticated {
  authenticated = yes;
  # Seems we need an "apply" section, even if empty, for symbols to work.
  apply {}
  symbols [
    "AUTH_USER"
  ]
}
```
Same result as you, If remove the apply clause, the symbol is never added.

**Reply by janedenone** [reply]
Thanks for the follow-up! I must admit I had already forgotten about this issue, but I am glad that it is reproducible.
